pkg/plugins/jaeger/router_test.go

```diff
@@ -0,0 +1,389 @@
+		req.Header.Set("x-kobs-plugin", "jaeger")
+		w := httptest.NewRecorder()
+
+		router.getOperations(w, req)
+
+		utils.AssertStatusEq(t, w, http.StatusOK)
+		utils.AssertJSONEq(t, w, `null`)
+	})
+}
+
+func TestGetTraces(t *testing.T) {
+	var newRouter = func(t *testing.T) (*instance.MockInstance, Router) {
+		ctrl := gomock.NewController(t)
+		mockInstance := instance.NewMockInstance(ctrl)
+		router := Router{chi.NewRouter(), []instance.Instance{mockInstance}}
+
+		return mockInstance, router
+	}
+
+	t.Run("should fail for invalid instance name", func(t *testing.T) {
+		i, router := newRouter(t)
+		i.EXPECT().GetName().Return("jaeger")
+
+		req, _ := http.NewRequestWithContext(context.Background(), http.MethodGet, "/traces", nil)
+		req.Header.Set("x-kobs-plugin", "invalidname")
+		w := httptest.NewRecorder()
+
+		router.getTraces(w, req)
+
+		utils.AssertStatusEq(t, w, http.StatusBadRequest)
+		utils.AssertJSONEq(t, w, `{"errors": ["Invalid plugin instance"]}`)
+	})
+
+	t.Run("should fail for invalid start time", func(t *testing.T) {
+		i, router := newRouter(t)
+		i.EXPECT().GetName().Return("jaeger")
+
+		req, _ := http.NewRequestWithContext(context.Background(), http.MethodGet, "/traces", nil)
+		req.Header.Set("x-kobs-plugin", "jaeger")
+		w := httptest.NewRecorder()
+
+		router.getTraces(w, req)
+
+		utils.AssertStatusEq(t, w, http.StatusBadRequest)
+		utils.AssertJSONEq(t, w, `{"errors": ["Failed to parse start time"]}`)
+	})
+
+	t.Run("should fail for invalid end time", func(t *testing.T) {
+		i, router := newRouter(t)
+		i.EXPECT().GetName().Return("jaeger")
+
+		req, _ := http.NewRequestWithContext(context.Background(), http.MethodGet, "/traces?timeStart=0", nil)
+		req.Header.Set("x-kobs-plugin", "jaeger")
+		w := httptest.NewRecorder()
+
+		router.getTraces(w, req)
+
+		utils.AssertStatusEq(t, w, http.StatusBadRequest)
+		utils.AssertJSONEq(t, w, `{"errors": ["Failed to parse end time"]}`)
+	})
+
+	t.Run("should fail when instance returns an error", func(t *testing.T) {
+		i, router := newRouter(t)
+		i.EXPECT().GetName().Return("jaeger")
+		i.EXPECT().GetTraces(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(nil, fmt.Errorf("unexpected error"))
+
+		req, _ := http.NewRequestWithContext(context.Background(), http.MethodGet, "/traces?timeStart=0&timeEnd=0", nil)
+		req.Header.Set("x-kobs-plugin", "jaeger")
+		w := httptest.NewRecorder()
+
+		router.getTraces(w, req)
+
+		utils.AssertStatusEq(t, w, http.StatusInternalServerError)
+		utils.AssertJSONEq(t, w, `{"errors": ["Failed to get traces"]}`)
+	})
+
+	t.Run("should return traces", func(t *testing.T) {
+		i, router := newRouter(t)
+		i.EXPECT().GetName().Return("jaeger")
+		i.EXPECT().GetTraces(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(nil, nil)
```

Comment:
We should use a non-trivial return value for the happy path and validate call (the params) to `GetTraces`.